### PR TITLE
Develop/alex grad2 2322

### PR DESCRIPTION
**Pull request by arybakov-cgi**

### CODE REVIEW

File: api/src/main/java/ca/bc/gov/educ/api/grad/report/service/CommonService.java [hunk 1]
```diff
@@ -845,6 +845,32 @@ private synchronized List<ReportGradStudentData> getReportGradStudentData(String
...
+    @Transactional
+    public Integer archiveSchoolReports(long batchId, List<String> schoolOfRecords, String reportType) {
+        Integer updatedReportsCount = 0;
+        Integer deletedReportsCount = 0;
+        Integer originalReportsCount = 0;
+        if(schoolOfRecords != null && !schoolOfRecords.isEmpty()) {
+            reportType = StringUtils.upperCase(StringUtils.endsWithIgnoreCase(reportType, "ARC") ? StringUtils.removeEndIgnoreCase(reportType, "ARC") : reportType);
+            String archivedReportType = StringUtils.upperCase(StringUtils.endsWith(reportType, "ARC") ? reportType : reportType + "ARC");
+            originalReportsCount += schoolReportsRepository.countBySchoolOfRecordsAndReportType(schoolOfRecords, reportType);
+            updatedReportsCount += schoolReportsRepository.archiveSchoolReports(schoolOfRecords, reportType, archivedReportType, batchId);
```
Comment (infstar) [inline api/src/main/java/ca/bc/gov/educ/api/grad/report/service/CommonService.java:865]: The logic here is right?

File: api/src/main/java/ca/bc/gov/educ/api/grad/report/service/CommonService.java [hunk 2]
```diff
@@ -845,6 +845,32 @@ private synchronized List<ReportGradStudentData> getReportGradStudentData(String
...
+            if(originalReportsCount.equals(updatedReportsCount)) {
```
Comment (infstar) [inline api/src/main/java/ca/bc/gov/educ/api/grad/report/service/CommonService.java:866]: The logic here is right?   It seems School reports are updated as Archived, and then deleted.

File: api/src/main/java/ca/bc/gov/educ/api/grad/report/repository/SchoolReportsRepository.java [hunk 3]
```diff
@@ -23,4 +24,18 @@ public interface SchoolReportsRepository extends JpaRepository<SchoolReportsEnti
...
+
+	@Query("select count(*) from SchoolReportsLightEntity c where c.reportTypeCode=:reportType")
+	Integer countByReportType(String reportType);
+
+	@Modifying
+	@Query(value="update SCHOOL_REPORT set REPORT_TYPE_CODE = :reportTypeTo, update_date = SYSDATE, update_user = 'Batch ' || :batchId || ' Archive Process' where school_of_record in (:schoolOfRecords) and REPORT_TYPE_CODE = :reportTypeFrom", nativeQuery=true)
+	Integer archiveSchoolReports(List<String> schoolOfRecords, String reportTypeFrom, String reportTypeTo, long batchId);
+
+	@Modifying
+	@Query(value="delete SCHOOL_REPORT where school_of_record in (:schoolOfRecords) and REPORT_TYPE_CODE = :reportType and UPDATE_DATE <= SYSDATE - 1", nativeQuery=true)
```
[the remark this answers is not in the document]
Comment (arybakov-cgi) [reply api/src/main/java/ca/bc/gov/educ/api/grad/report/repository/SchoolReportsRepository.java:38]: Logic is correct. We delete OLD archived reports, not just created. It corresponds to your question about CommonService.

[the remark this answers is not in the document]
Comment (arybakov-cgi) [reply api/src/main/java/ca/bc/gov/educ/api/grad/report/repository/SchoolReportsRepository.java:38]: just delete SCHOOL_REPORT is fine

**Review by infstar — APPROVED** [review]

**Pull request merged**